plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -912,12 +912,27 @@ public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHan
     @Override
     public void createSchema(ConnectorSession session, String schemaName)
     {
+        try (Connection connection = connectionFactory.openConnection(session)) {
+            if (connection.getMetaData().storesUpperCaseIdentifiers()) {
```

Comment:
That is a pity that we need to open a connection for that. 
Is it possibly that depending on `session` it could return different value? Maybe it should be simply cached and reused.

Reply:
I'm not sure we should be concerned about this. `execute(session, query)` also gets a connection. I can change it to reuse the same connection. I'm worried I wouldn't be able to prove there are no side effects if we start caching things.